agent/metrics/instancehealth/instancehealth.go

```diff
@@ -26,16 +30,28 @@ func (gc *genericMetric) IncrementCallCount() {
 
 func (gc *genericMetric) RecordError(errorMessage errors.NamedError) {
 	atomic.AddInt64(&gc.errors, 1)
-	gc.errorMessage.Store(errorMessage.ErrorName() + ": " + errorMessage.Error())
+	atomic.AddInt64(&gc.calls, 1)
```

Comment:
What's the reason we update the call count again here?

Reply:
That was me misreading the interface comment...I will remove this